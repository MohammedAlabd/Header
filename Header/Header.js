import React from "react";
import MainNavBar from "./MainNavBar";
import SubNavBar from "./SubNavBar";
import { useLocation } from "react-router-dom";

export default function Header() {
  const location = useLocation();

  const pages = [
    { name: "Home", url: "", list: false },
    {
      name: "Services",
      url: "services",
      list: [
        { name: "Individuals", url: "individuals" },
        { name: "Organizations", url: "organizations" },
        { name: "Academic", url: "academic" },
        { name: "Translation", url: "translation" },
      ],
    },
    { name: "About Us", url: "about", list: false },
    { name: "Blog", url: "blog", list: false },
    { name: "Contact Us", url: "contact", list: false },
    { name: "Shop", url: "shop", list: false },
  ];

  const mainNavbarHeight = 88;

  const currentPageName = location.pathname.split("/")[1];
  const currentPageObject = pages.find((page) => page.url === currentPageName);
  return (
    <>
      <MainNavBar
        mainNavbarHeight={mainNavbarHeight}
        currentPagePath={
          location.pathname.includes("/blogs") ? "/POSTS" : location.pathname
        }
        pages={pages}
      />
      {currentPageObject && currentPageObject.list ? (
        <SubNavBar
          mainNavbarHeight={mainNavbarHeight}
          currentPagePath={location.pathname}
          pageObject={currentPageObject}
        />
      ) : null}
    </>
  );
}
