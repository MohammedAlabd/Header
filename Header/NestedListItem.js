import React, { useState } from "react";
import { makeStyles } from "@material-ui/core/styles";
import ExpandLess from "@material-ui/icons/ExpandLess";
import ExpandMore from "@material-ui/icons/ExpandMore";
import Collapse from "@material-ui/core/Collapse";
import { List, ListItem, Typography } from "@material-ui/core";
import { Link } from "react-router-dom";

const useStyles = makeStyles((theme) => ({
  nested: {
    paddingLeft: theme.spacing(4),
  },
  textDecoration: {
    textDecoration: "none",
  },
}));

export default function NestedListItem({
  page,
  currentPagePath,
  handleDrawerClose,
}) {
  const classes = useStyles();

  const [isNestedListOpen, setIsNestedListOpen] = useState(false);

  const handlePageSelect = () => {
    handleToggleNestedList();
    handleDrawerClose();
  };

  const handleToggleNestedList = () => {
    setIsNestedListOpen(!isNestedListOpen);
  };

  // this regex used to search for the page name in the url
  const listName = new RegExp(`/${page.url}`, "i");
  const isListSelected = currentPagePath.search(listName) === 0;

  return (
    <>
      <ListItem
        button
        data-testid="openMenuButton"
        onClick={handleToggleNestedList}
      >
        <Typography
          color={isListSelected ? "primary" : "secondary"}
          variant="subtitle1"
        >
          {page.name.toLocaleUpperCase()}
        </Typography>
        {isNestedListOpen ? (
          <ExpandLess color={isListSelected ? "primary" : "secondary"} />
        ) : (
          <ExpandMore color={isListSelected ? "primary" : "secondary"} />
        )}
      </ListItem>
      <Collapse in={isNestedListOpen} timeout="auto" unmountOnExit>
        <List data-testid="nestedMenu" component="div" disablePadding>
          {page.list.map((listItem) => {
            const isPageSelected =
              currentPagePath === `/${page.url}/${listItem.url}`;
            return (
              <Link
                key={listItem.url}
                to={`/${page.url}/${listItem.url}`}
                className={classes.textDecoration}
                onClick={handlePageSelect}
              >
                <ListItem button>
                  <Typography
                    color={isPageSelected ? "primary" : "secondary"}
                    variant="subtitle1"
                  >
                    {listItem.name.toLocaleUpperCase()}
                  </Typography>
                </ListItem>
              </Link>
            );
          })}
        </List>
      </Collapse>
    </>
  );
}
