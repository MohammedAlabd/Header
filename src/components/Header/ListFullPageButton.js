import React, { useState } from "react";
import ExpandLess from "@material-ui/icons/ExpandLess";
import ExpandMore from "@material-ui/icons/ExpandMore";
import { Button, Menu, MenuItem, Typography } from "@material-ui/core";
import { NavLink } from "react-router-dom";
import { makeStyles } from "@material-ui/core/styles";

const useStyles = makeStyles(() => ({
  textDecoration: {
    textDecoration: "none",
  },
}));

export default function ListFullPage({ page, currentPagePath }) {
  const classes = useStyles();

  const [anchorEl, setAnchorEl] = useState(null);

  const handleOpenMenu = (event) => {
    setAnchorEl(event.currentTarget);
  };

  const handleCloseMenu = () => {
    setAnchorEl(null);
  };

  // this regex used to search for the page name in the url
  const listName = new RegExp(`/${page.url}`, "i");
  const isListSelected = currentPagePath.search(listName) === 0;

  return (
    <>
      <Button
        data-testid="openMenuButton"
        color={isListSelected ? "primary" : "secondary"}
        onClick={handleOpenMenu}
      >
        <Typography
          color={isListSelected ? "primary" : "secondary"}
          variant="subtitle1"
        >
          {page.name}
        </Typography>
        {Boolean(anchorEl) ? <ExpandLess /> : <ExpandMore />}
      </Button>
      <Menu
        data-testid="menu"
        anchorEl={anchorEl}
        keepMounted
        open={Boolean(anchorEl)}
        onClose={handleCloseMenu}
      >
        {page.list.map((listItem) => {
          const isPageSelected =
            currentPagePath === `/${page.url}/${listItem.url}`;

          return (
            <NavLink
              onClick={handleCloseMenu}
              to={`/${page.url}/${listItem.url}`}
              className={classes.textDecoration}
              key={listItem.url}
            >
              <MenuItem>
                <Typography
                  variant="subtitle1"
                  color={isPageSelected ? "primary" : "secondary"}
                >
                  {listItem.name.toLocaleUpperCase()}
                </Typography>
              </MenuItem>
            </NavLink>
          );
        })}
      </Menu>
    </>
  );
}
