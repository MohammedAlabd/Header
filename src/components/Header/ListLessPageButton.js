import React from "react";
import { Button, Typography } from "@material-ui/core";
import { NavLink } from "react-router-dom";
import { makeStyles } from "@material-ui/core/styles";

const useStyles = makeStyles((theme) => ({
  textDecoration: {
    textDecoration: "none",
  },
  activeLink: {
    color: theme.palette.primary.main,
  },
}));

export default function ListLessPage({ page, currentPagePath }) {
  const classes = useStyles();

  const isPageSelected = currentPagePath === `/${page.url}`;

  return (
    <NavLink to={`/${page.url}`} className={classes.textDecoration}>
      <Button color={isPageSelected ? "primary" : "secondary"}>
        <Typography
          color={isPageSelected ? "primary" : "secondary"}
          variant="subtitle1"
        >
          {page.name}
        </Typography>
      </Button>
    </NavLink>
  );
}
