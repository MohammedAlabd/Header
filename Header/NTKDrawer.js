import React from "react";
import { makeStyles } from "@material-ui/core/styles";
import ExpandLess from "@material-ui/icons/ExpandLess";
import { Drawer, List, IconButton } from "@material-ui/core";
import NestedListItem from "./NestedListItem";
import SingleListItem from "./SingleListItem";

const drawerWidth = "auto";

const useStyles = makeStyles((theme) => ({
  drawer: {
    width: drawerWidth,
    flexShrink: 0,
  },
  drawerPaper: {
    width: drawerWidth,
  },
  drawerHeader: {
    display: "flex",
    alignItems: "center",
    padding: theme.spacing(0, 1),
    // necessary for content to be below app bar
    ...theme.mixins.toolbar,
    justifyContent: "flex-end",
  },
}));

export default function NTKDrawer({
  pages,
  isDrawerOpen,
  handleDrawerClose,
  currentPagePath,
}) {
  const classes = useStyles();

  return (
    <Drawer
      className={classes.drawer}
      variant="persistent"
      anchor="top"
      open={isDrawerOpen}
      classes={{ paper: classes.drawerPaper }}
    >
      <div className={classes.drawerHeader}>
        <IconButton onClick={handleDrawerClose}>
          <ExpandLess />
        </IconButton>
      </div>
      <List>
        {pages.map((page) => {
          if (page.list) {
            return (
              <NestedListItem
                key={page.url}
                handleDrawerClose={handleDrawerClose}
                currentPagePath={currentPagePath}
                page={page}
              />
            );
          } else {
            return (
              <SingleListItem
                key={page.url}
                handleDrawerClose={handleDrawerClose}
                currentPagePath={currentPagePath}
                page={page}
              />
            );
          }
        })}
      </List>
    </Drawer>
  );
}
