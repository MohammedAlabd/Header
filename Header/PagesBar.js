import React from "react";
import { Grid } from "@material-ui/core";
import ListFullPageButton from "./ListFullPageButton";
import ListLessPageButton from "./ListLessPageButton";

export default function PagesBar({ pages, currentPagePath }) {
  // we don't need to render the home page because we already have the logo
  const pagesToBeRendered = pages.slice(1);

  return (
    <Grid container spacing={3}>
      {pagesToBeRendered.map((page) => {
        if (page.list) {
          return (
            <Grid item key={page.url}>
              <ListFullPageButton
                page={page}
                currentPagePath={currentPagePath}
              />
            </Grid>
          );
        } else {
          return (
            <Grid item key={page.url}>
              <ListLessPageButton
                page={page}
                currentPagePath={currentPagePath}
              />
            </Grid>
          );
        }
      })}
    </Grid>
  );
}
