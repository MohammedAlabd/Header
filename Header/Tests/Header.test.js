import React from "react";
import Header from "../Header";
import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";

it("matches snapshot", () => {
  const tree = render(
    <MemoryRouter>
      <Header />
    </MemoryRouter>
  );
  expect(tree).toMatchSnapshot();
});
