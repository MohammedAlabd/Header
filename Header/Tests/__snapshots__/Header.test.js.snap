// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="makeStyles-root-1"
      >
        <header
          class="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary makeStyles-appBar-2  mui-fixed MuiPaper-elevation4"
        >
          <div
            class="MuiToolbar-root MuiToolbar-regular makeStyles-container-9 makeStyles-container-11 MuiToolbar-gutters"
          >
            <a
              aria-current="page"
              class="makeStyles-logo-8 active"
              href="/"
            >
              <img
                alt="the Logo of Natakalam"
                src="logo.png"
                width="35px"
              />
            </a>
            <div
              class="makeStyles-responsiveNav-7"
            >
              <div
                class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3"
              >
                <div
                  class="MuiGrid-root MuiGrid-item"
                >
                  <button
                    class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSecondary"
                    data-testid="openMenuButton"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="MuiButton-label"
                    >
                      <h6
                        class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-colorSecondary"
                      >
                        Services
                      </h6>
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                        />
                      </svg>
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item"
                >
                  <a
                    class="makeStyles-textDecoration-234"
                    href="/about"
                  >
                    <button
                      class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSecondary"
                      tabindex="0"
                      type="button"
                    >
                      <span
                        class="MuiButton-label"
                      >
                        <h6
                          class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-colorSecondary"
                        >
                          About Us
                        </h6>
                      </span>
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </button>
                  </a>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item"
                >
                  <a
                    class="makeStyles-textDecoration-234"
                    href="/blog"
                  >
                    <button
                      class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSecondary"
                      tabindex="0"
                      type="button"
                    >
                      <span
                        class="MuiButton-label"
                      >
                        <h6
                          class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-colorSecondary"
                        >
                          Blog
                        </h6>
                      </span>
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </button>
                  </a>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item"
                >
                  <a
                    class="makeStyles-textDecoration-234"
                    href="/contact"
                  >
                    <button
                      class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSecondary"
                      tabindex="0"
                      type="button"
                    >
                      <span
                        class="MuiButton-label"
                      >
                        <h6
                          class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-colorSecondary"
                        >
                          Contact Us
                        </h6>
                      </span>
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </button>
                  </a>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item"
                >
                  <a
                    class="makeStyles-textDecoration-234"
                    href="/shop"
                  >
                    <button
                      class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSecondary"
                      tabindex="0"
                      type="button"
                    >
                      <span
                        class="MuiButton-label"
                      >
                        <h6
                          class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-colorSecondary"
                        >
                          Shop
                        </h6>
                      </span>
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </button>
                  </a>
                </div>
              </div>
            </div>
            <h6
              class="MuiTypography-root makeStyles-responsivePagePath-10 MuiTypography-subtitle1 MuiTypography-colorPrimary"
            />
            <button
              aria-label="open drawer"
              class="MuiButtonBase-root MuiIconButton-root makeStyles-responsiveDrawer-6  MuiIconButton-colorSecondary MuiIconButton-edgeEnd"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </header>
        <div
          class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
        />
        <div
          class="MuiDrawer-root MuiDrawer-docked makeStyles-drawer-245"
        >
          <div
            class="MuiPaper-root MuiDrawer-paper makeStyles-drawerPaper-246 MuiDrawer-paperAnchorTop MuiDrawer-paperAnchorDockedTop MuiPaper-elevation0"
            style="visibility: hidden; webkit-transform: translateY(-0px); transform: translateY(-0px);"
          >
            <div
              class="makeStyles-drawerHeader-247"
            >
              <button
                class="MuiButtonBase-root MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 8l-6 6 1.41 1.41L12 10.83l4.59 4.58L18 14z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
            <ul
              class="MuiList-root MuiList-padding"
            >
              <a
                class="makeStyles-textDecoration-264"
                href="/"
              >
                <div
                  aria-disabled="false"
                  class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
                  role="button"
                  tabindex="0"
                >
                  <h6
                    class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-colorPrimary"
                  >
                    HOME
                  </h6>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </div>
              </a>
              <div
                aria-disabled="false"
                class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
                data-testid="openMenuButton"
                role="button"
                tabindex="0"
              >
                <h6
                  class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-colorSecondary"
                >
                  SERVICES
                </h6>
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-colorSecondary"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root"
                />
              </div>
              <a
                class="makeStyles-textDecoration-264"
                href="/about"
              >
                <div
                  aria-disabled="false"
                  class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
                  role="button"
                  tabindex="0"
                >
                  <h6
                    class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-colorSecondary"
                  >
                    ABOUT US
                  </h6>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </div>
              </a>
              <a
                class="makeStyles-textDecoration-264"
                href="/blog"
              >
                <div
                  aria-disabled="false"
                  class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
                  role="button"
                  tabindex="0"
                >
                  <h6
                    class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-colorSecondary"
                  >
                    BLOG
                  </h6>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </div>
              </a>
              <a
                class="makeStyles-textDecoration-264"
                href="/contact"
              >
                <div
                  aria-disabled="false"
                  class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
                  role="button"
                  tabindex="0"
                >
                  <h6
                    class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-colorSecondary"
                  >
                    CONTACT US
                  </h6>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </div>
              </a>
              <a
                class="makeStyles-textDecoration-264"
                href="/shop"
              >
                <div
                  aria-disabled="false"
                  class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
                  role="button"
                  tabindex="0"
                >
                  <h6
                    class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-colorSecondary"
                  >
                    SHOP
                  </h6>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </div>
              </a>
            </ul>
          </div>
        </div>
      </div>
    </div>
    <div
      aria-hidden="true"
      class="MuiPopover-root"
      data-testid="menu"
      role="presentation"
      style="position: fixed; z-index: 1300; right: 0px; bottom: 0px; top: 0px; left: 0px; visibility: hidden;"
    >
      <div
        data-test="sentinelStart"
        tabindex="0"
      />
      <div
        class="MuiPaper-root MuiMenu-paper MuiPopover-paper MuiPaper-elevation8 MuiPaper-rounded"
        style="opacity: 0; transform: scale(0.75, 0.5625); visibility: hidden;"
        tabindex="-1"
      >
        <ul
          class="MuiList-root MuiMenu-list MuiList-padding"
          role="menu"
          tabindex="-1"
        >
          <a
            class="makeStyles-textDecoration-158"
            href="/services/individuals"
            tabindex="0"
          >
            <li
              aria-disabled="false"
              class="MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
              role="menuitem"
              tabindex="-1"
            >
              <h6
                class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-colorSecondary"
              >
                INDIVIDUALS
              </h6>
              <span
                class="MuiTouchRipple-root"
              />
            </li>
          </a>
          <a
            class="makeStyles-textDecoration-158"
            href="/services/organizations"
          >
            <li
              aria-disabled="false"
              class="MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
              role="menuitem"
              tabindex="-1"
            >
              <h6
                class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-colorSecondary"
              >
                ORGANIZATIONS
              </h6>
              <span
                class="MuiTouchRipple-root"
              />
            </li>
          </a>
          <a
            class="makeStyles-textDecoration-158"
            href="/services/academic"
          >
            <li
              aria-disabled="false"
              class="MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
              role="menuitem"
              tabindex="-1"
            >
              <h6
                class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-colorSecondary"
              >
                ACADEMIC
              </h6>
              <span
                class="MuiTouchRipple-root"
              />
            </li>
          </a>
          <a
            class="makeStyles-textDecoration-158"
            href="/services/translation"
          >
            <li
              aria-disabled="false"
              class="MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
              role="menuitem"
              tabindex="-1"
            >
              <h6
                class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-colorSecondary"
              >
                TRANSLATION
              </h6>
              <span
                class="MuiTouchRipple-root"
              />
            </li>
          </a>
        </ul>
      </div>
      <div
        data-test="sentinelEnd"
        tabindex="0"
      />
    </div>
  </body>,
  "container": <div>
    <div
      class="makeStyles-root-1"
    >
      <header
        class="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary makeStyles-appBar-2  mui-fixed MuiPaper-elevation4"
      >
        <div
          class="MuiToolbar-root MuiToolbar-regular makeStyles-container-9 makeStyles-container-11 MuiToolbar-gutters"
        >
          <a
            aria-current="page"
            class="makeStyles-logo-8 active"
            href="/"
          >
            <img
              alt="the Logo of Natakalam"
              src="logo.png"
              width="35px"
            />
          </a>
          <div
            class="makeStyles-responsiveNav-7"
          >
            <div
              class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3"
            >
              <div
                class="MuiGrid-root MuiGrid-item"
              >
                <button
                  class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSecondary"
                  data-testid="openMenuButton"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="MuiButton-label"
                  >
                    <h6
                      class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-colorSecondary"
                    >
                      Services
                    </h6>
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item"
              >
                <a
                  class="makeStyles-textDecoration-234"
                  href="/about"
                >
                  <button
                    class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSecondary"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="MuiButton-label"
                    >
                      <h6
                        class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-colorSecondary"
                      >
                        About Us
                      </h6>
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                </a>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item"
              >
                <a
                  class="makeStyles-textDecoration-234"
                  href="/blog"
                >
                  <button
                    class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSecondary"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="MuiButton-label"
                    >
                      <h6
                        class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-colorSecondary"
                      >
                        Blog
                      </h6>
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                </a>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item"
              >
                <a
                  class="makeStyles-textDecoration-234"
                  href="/contact"
                >
                  <button
                    class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSecondary"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="MuiButton-label"
                    >
                      <h6
                        class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-colorSecondary"
                      >
                        Contact Us
                      </h6>
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                </a>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item"
              >
                <a
                  class="makeStyles-textDecoration-234"
                  href="/shop"
                >
                  <button
                    class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSecondary"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="MuiButton-label"
                    >
                      <h6
                        class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-colorSecondary"
                      >
                        Shop
                      </h6>
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                </a>
              </div>
            </div>
          </div>
          <h6
            class="MuiTypography-root makeStyles-responsivePagePath-10 MuiTypography-subtitle1 MuiTypography-colorPrimary"
          />
          <button
            aria-label="open drawer"
            class="MuiButtonBase-root MuiIconButton-root makeStyles-responsiveDrawer-6  MuiIconButton-colorSecondary MuiIconButton-edgeEnd"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </header>
      <div
        class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
      />
      <div
        class="MuiDrawer-root MuiDrawer-docked makeStyles-drawer-245"
      >
        <div
          class="MuiPaper-root MuiDrawer-paper makeStyles-drawerPaper-246 MuiDrawer-paperAnchorTop MuiDrawer-paperAnchorDockedTop MuiPaper-elevation0"
          style="visibility: hidden; webkit-transform: translateY(-0px); transform: translateY(-0px);"
        >
          <div
            class="makeStyles-drawerHeader-247"
          >
            <button
              class="MuiButtonBase-root MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 8l-6 6 1.41 1.41L12 10.83l4.59 4.58L18 14z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
          <ul
            class="MuiList-root MuiList-padding"
          >
            <a
              class="makeStyles-textDecoration-264"
              href="/"
            >
              <div
                aria-disabled="false"
                class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
                role="button"
                tabindex="0"
              >
                <h6
                  class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-colorPrimary"
                >
                  HOME
                </h6>
                <span
                  class="MuiTouchRipple-root"
                />
              </div>
            </a>
            <div
              aria-disabled="false"
              class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
              data-testid="openMenuButton"
              role="button"
              tabindex="0"
            >
              <h6
                class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-colorSecondary"
              >
                SERVICES
              </h6>
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-colorSecondary"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root"
              />
            </div>
            <a
              class="makeStyles-textDecoration-264"
              href="/about"
            >
              <div
                aria-disabled="false"
                class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
                role="button"
                tabindex="0"
              >
                <h6
                  class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-colorSecondary"
                >
                  ABOUT US
                </h6>
                <span
                  class="MuiTouchRipple-root"
                />
              </div>
            </a>
            <a
              class="makeStyles-textDecoration-264"
              href="/blog"
            >
              <div
                aria-disabled="false"
                class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
                role="button"
                tabindex="0"
              >
                <h6
                  class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-colorSecondary"
                >
                  BLOG
                </h6>
                <span
                  class="MuiTouchRipple-root"
                />
              </div>
            </a>
            <a
              class="makeStyles-textDecoration-264"
              href="/contact"
            >
              <div
                aria-disabled="false"
                class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
                role="button"
                tabindex="0"
              >
                <h6
                  class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-colorSecondary"
                >
                  CONTACT US
                </h6>
                <span
                  class="MuiTouchRipple-root"
                />
              </div>
            </a>
            <a
              class="makeStyles-textDecoration-264"
              href="/shop"
            >
              <div
                aria-disabled="false"
                class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
                role="button"
                tabindex="0"
              >
                <h6
                  class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-colorSecondary"
                >
                  SHOP
                </h6>
                <span
                  class="MuiTouchRipple-root"
                />
              </div>
            </a>
          </ul>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
